source/common/http/conn_manager_impl.cc

```diff
@@ -661,7 +662,7 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(HeaderMapPtr&& headers,
   // when the allow_absolute_url flag is enabled on the HCM.
   // https://tools.ietf.org/html/rfc7230#section-5.3 We also need to check for the existence of
   // :path because CONNECT does not have a path, and we don't support that currently.
-  if (!request_headers_->Path() || request_headers_->Path()->value().c_str()[0] != '/') {
+  if (!request_headers_->Path() || request_headers_->Path()->value().getStringView()[0] != '/') {
```

Comment:
In a world where the string does not have a trailing NUL I think this indexing is now correct and we have to also check for not-empty? Can you either fix or put in a TODO for when you get rid of the trailing NUL? Can you double check that we don't have more cases like this that might subtly break?

Reply:
I think it's OK in the current state of the world, but definitely an issue if we remove the trailing NUL. I'll fix it now.